driver-legacy/src/main/com/mongodb/Cursor.java

```diff
@@ -27,6 +27,14 @@
  */
 public interface Cursor extends Iterator<DBObject>, Closeable {
 
+    /**
+     * Gets the number of results available locally without blocking,which may be 0, or 0 when the cursor is exhausted or closed.
```

Comment:
(Typo? `which may be 0, or 0 when...`)

Reply:
Not sure what you mean.  (There is a missing space after the comma, but I don't think that's what you're referring to.) Can you write the sentence the way you think it should be? 

Note that I cribbed this from `java.io.InputStream#available`, which says:

> Returns an estimate of the number of bytes that can be read (or skipped over) from this input stream without blocking, which may be 0, or 0 when end of stream is detected.

Reply:
Ah, it's just the language is a little confusing - think I get it now. It could return 0 in two scenarios: because there are no results available locally, or because the cursor is closed (the first clause covers the cursor being exhausted, I think?)

Don't feel strongly, just found it a bit tricky to parse.

Reply:
No, the second clause covers both exhausted and closed.  The first clause covers the case where the next call to next/hasNext would have to do I/O to fetch more results, i.e. by executing a getMore, as opposed to the non-zero case where there are still documents in the currently fetched batch (either from the initial find/aggregate or the previous getMore)

Reply:
I agree that the _"may be 0, or 0"_ part is difficult to understand.